modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/result/AbstractReturnDelegate.java

```diff
@@ -293,13 +302,36 @@ protected MediaType resolveMediaType(Object value, MediaType contextMimeType, Ch
     return contextMimeType.withCharset(contextEncoding);
   }
 
+  private Runnable incrementActiveComponent(ExecutionContextAdapter executionContext) {
+    Optional<ConfigurationInstance> config = executionContext.getConfiguration();
+    return config
+        .<Runnable>map(configurationInstance -> () -> tryToMutateConfigurationStats(configurationInstance,
+                                                                                    (MutableConfigurationStats::addActiveComponent)))
+        .orElseGet(() -> () -> {
+        });
+  }
+
+  private Runnable decrementActiveComponent(ExecutionContextAdapter executionContext) {
```

Comment:
By reading the name of these methods, it seems that it changes the counter, I would change them to reflect that the runnable that modifies it is returned.